### PR TITLE
Updating user docs on Notification settings

### PR DESCRIPTION
**Pull request by psatyal**
Rewrote the Notification Settings docs to reflect:

- New organisation and phrasing
- Addition of date alerts

To do:
- Update the bit about project-specific notifications
- Add updated screenshots

### CODE REVIEW

**Comment by psatyal** [comment]
Hi @as-op, as always, could you please verify if relative paths are correct? Thank you.

File: docs/user-guide/notifications/notification-settings/README.md [hunk 1]
```diff
@@ -5,26 +5,66 @@ sidebar_navigation:
...
-By default you do not receive any notifications about your own changes. In addition to the in-app notifications, you will also get a once-a-day summary of all notifications as **[Email reminders](../../../getting-started/my-account#email-reminders)**.
+You cannot disable notifications for when you are mentioned (since the goal of mentioning you is to get your attention) or for work packages that you are watching. For the latter, you may disable further notifications simply by unwatching those work package.
+
+> Info: Modifying these settings might result in your missing updates and changes that are relevant to you. We do not recommend changing them unless you are absolutely certain of the consequences.
+
+## Date alerts
+
+Starting 12.4, Open Project offers notification for date alerts. Please note that this is an Enterprise feature.
+
+> **Note**: Date alerts are a Premium Feature and can only be used with [Enterprise cloud](../../enterprise-guide/enterprise-cloud-guide/) or [Enterprise on-premises](../../enterprise-guide/enterprise-on-premises-guide/). An upgrade from the free Community Edition is easy and helps support OpenProject.
```
Comment (as-op) [inline docs/user-guide/notifications/notification-settings/README.md]: The links to the enterprise guides need to go one more level up:
`> **Note**: Date alerts are a Premium Feature and can only be used with [Enterprise cloud](../../../enterprise-guide/enterprise-cloud-guide/) or [Enterprise on-premises](../../../enterprise-guide/enterprise-on-premises-guide/). An upgrade from the free Community Edition is easy and helps support OpenProject.
`

**Review by as-op — CHANGES_REQUESTED** [review]

File: docs/user-guide/notifications/notification-settings/README.md (hunk 1 above)
Comment (psatyal) [reply docs/user-guide/notifications/notification-settings/README.md]: Thank you @as-op, fixed!

**Review by psatyal — COMMENTED** [review]
Fixed.

**Review by as-op — APPROVED** [review]

**Comment by psatyal** [comment]
Linking to work package for history: https://community.openproject.org/projects/openproject/work_packages/44729/activity

**Pull request merged**